packages/integration-sdk-runtime/src/execution/__tests__/dependencyGraph.test.ts

```diff
@@ -622,6 +630,156 @@ describe('executeStepDependencyGraph', () => {
     expect(spyB).toHaveBeenCalledBefore(spyC);
   });
 
+  test('should mark steps failed executionHandlers with status FAILURE a dependent steps with status PARTIAL_SUCCESS_DUE_TO_DEPENDENCY_FAILURE when step upload fails', async () => {
```

Comment:
I think there are typos here? Hard to understand.

```suggestion
  test('should mark steps with failed executionHandlers with status FAILURE and dependent steps with status PARTIAL_SUCCESS_DUE_TO_DEPENDENCY_FAILURE when step upload fails', async () => {
```

Reply:
I guess this also doesn't seem like a test that's relevant to this PR. The scope seems to be about dependencies here, although the functionality is about creating a failing uploader. Consider trimming this, breaking it apart, or making the test name clearer.

Reply:
Will fix the grammar. Thanks for the suggestion. The test is relevant to the overall changes though. The test validates that our uploads respect the existing behavior of our internal dependency graph.